weaviate/connect/connection.py

```diff
@@ -661,13 +649,16 @@ def get(
         else:
             request_url = self.url + self._api_version_path + path
 
-        return self._session.get(
-            url=request_url,
-            headers=self._get_request_header(),
-            timeout=self._timeout_config,
-            params=params,
-            proxies=self._proxies,
-        )
+        try:
+            return self._session.get(
+                url=request_url,
+                headers=self._get_request_header(),
+                timeout=self._timeout_config,
+                params=params,
+                proxies=self._proxies,
+            )
+        except requests.exceptions.ConnectionError as e:
+            raise WeaviateStartUpError(f"Could not connect to Weaviate:{e.strerror}.") from e
```

Comment:
I like having this check here. Should we also have it within the other REST methods and maybe also wrapping the gRPC calls?

Reply:
Makes sense!